apps/emqx_authz/src/emqx_authz_http.erl

```diff
@@ -121,30 +141,65 @@ serialize_body(<<"application/json">>, Body) ->
 serialize_body(<<"application/x-www-form-urlencoded">>, Body) ->
     query_string(Body).
 
-replvar(Str0, PubSub, Topic,
-        #{username := Username,
-          clientid := Clientid,
-          peerhost := IpAddress,
-          protocol := Protocol,
-          mountpoint := Mountpoint
-         }) when is_list(Str0);
-                 is_binary(Str0) ->
+
+replvar_deep(Map, PubSub, Topic, Vars, VarEncode) when is_map(Map) ->
+    maps:map(
+      fun(_Key, Value) ->
```

Comment:
Ignoring the key here looks surprising. It would be nice to have a comment explaining why it's discarded

Reply:
Mapping only values replicates the logic of other backends, like `https://github.com/emqx/emqx/blob/master/apps/emqx_authz/src/emqx_authz_mongodb.erl#L95` where we are unlikely to have substitutions in the _keys_ of a selector.

This var substitution code is a subject for refactoring, but yes, now we better map keys too, to match the originally intended behaviour.

Reply:
✅